multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/executor/TimeoutAsyncTaskExecutor.java

```diff
@@ -0,0 +1,33 @@
+package org.cloudfoundry.multiapps.controller.process.executor;
+
+import org.flowable.common.engine.impl.async.DefaultAsyncTaskExecutor;
+
+import java.util.concurrent.Future;
+
+public class TimeoutAsyncTaskExecutor extends DefaultAsyncTaskExecutor {
+
+    private final CancelExpiredTasksThread cancelExpiredTasksThread;
+
+    public TimeoutAsyncTaskExecutor(long timeoutInMillis, long waitTimeInMillis) {
+        executorNeedsShutdown = true;
+        cancelExpiredTasksThread = new CancelExpiredTasksThread(timeoutInMillis, waitTimeInMillis);
+    }
+
+    @Override
+    public void execute(Runnable task) {
+        Future<?> f = executorService.submit(task);
+        cancelExpiredTasksThread.addTask(f);
+    }
+
+    @Override
+    public void start() {
+        super.start();
+        cancelExpiredTasksThread.start();
```

Comment:
Shouldn't any custom logic be executed before any calls to methods of the parent class?

Reply:
in this case it does not matter - it can be switched